src/mono/mono/mini/mini.c

```diff
@@ -3261,6 +3261,18 @@ mini_method_compile (MonoMethod *method, guint32 opts, JitFlags flags, int parts
 		cfg->gsctx_context = context;
 
 		cfg->gsharedvt = TRUE;
+		if (cfg->llvm_only) {
+			/*
+			 * Enable a minimal version of gsharedvt where only
+			 * methods which don't really depend on their generic
+			 * arguments are supported, like List<T>.get_Count ().
+			 * - no signatures or locals with variable length types
```

Comment:
Would it be possible to extend this so that if the function has a local or argument of generic instance of a reference type e.g. `List<T>` - those aren't really variable sized.

You're already sort of supporting this because `this` "uses" the gsharedvt generic param.


Actually, do we need to check the type of `this` in `mini_is_gsharedvt_signature`?  it's always boxed if we had a generic struct, right?

Reply:
Fixed.